sources/sql_database/__init__.py

```diff
@@ -111,7 +111,7 @@ def sql_database(
             name=table.name,
             primary_key=get_primary_key(table),
             spec=SqlDatabaseTableConfiguration,
-            columns=table_to_columns(table, reflection_level, type_adapter_callback),
+            # columns hint will be set at runtime, after included_columns setting has been resolved
```

Comment:
@rudolfix is there any sense in trying to resolve the config there before resource is called? The column hints are now added with filtered columns in `table_rows` (same as with deferred reflection mode before)

Reply:
hmmm `sql_table` actually resolves config before we create resource. maybe we just call it here and yield what it returns?

Reply:
Hmm maybe that would work. I'll try it.

Reply:
Ok re-using `sql_table` resource now, seems to work fine and reduces duplication a bit.